Core/src/IfcGeometryConverter/PlacementConverter.h

```diff
@@ -757,6 +758,93 @@ namespace OpenInfraPlatform {
 					return object_placement_matrix;
                 }
 
+				carve::geom::vector<3> convertIfcVirtualGridIntersection(
+					const EXPRESSReference<typename IfcEntityTypesT::IfcVirtualGridIntersection> &intersection
+				) const throw(...)
+				{
+					if (intersection.expired())
+						throw oip::ReferenceExpiredException(intersection);
+
+
+					std::vector<carve::geom::vector<2>> Xaxis;
+					std::vector<carve::geom::vector<2>> Yaxis;
+					std::vector<carve::geom::vector<2>> Xsegment_start_points;
+					std::vector<carve::geom::vector<2>> Ysegment_start_points;
+
+
+					CurveConverterT<IfcEntityTypesT> grid_conv(GeomSettings(), UnitConvert(), this);
+
+					grid_conv.convertIfcCurve2D(intersection->IntersectingAxes[0], Xaxis, Xsegment_start_points);
+					grid_conv.convertIfcCurve2D(intersection->IntersectingAxes[1], Yaxis, Ysegment_start_points);
+					
+					std::vector<carve::geom::vector<2>> intersection_point;
+
+					for (int i = 0; i < Xaxis.size(); i++) {
+						X_Axis_Position = XAxis[i];
+						X_Axis_2Position = XAxis[i+1];
+						for (int j = 0; j <= Yaxis.size(); j++) {
+							Y_Axis_Position = YAxis[i];
+							Y_Axis_2Position = YAxis[i+1];
+							if (GeomUtils::LineSegmentToLineIntersection(X_Axis_2Position, X_Axis_Position, Y_Axis_2Position, Y_Axis_Position, intersection_point)) {
+								goto stop;

```

Comment:
`goto` is bad programming. Yes, it is possible, but you need to exit both loops differently.

E.g.: set `j=Yaxis.size()` and `i=Xaxis.size()` - in this way, both loops will terminate.

Reply:
Resolved: c4244fc